### ISSUE TITLE
Android push notifications failing to launch app after SDK version bump

### ISSUE THREAD

**Issue by Ricardinh0** [Appboy/appboy-cordova-sdk#87]
## Context

Hello 👋 

We recently updated our `appboy-cordova-sdk` version from `2.24.0` to `2.30.1`. This version bump was done in parallel with bumping our apps version of Iconic `capacitor` from v3 to v4. 

The main change that came out of this regarding our Braze integration was to update the `AndroidManifest.xml` to using the new `com.braze.push.BrazeFirebaseMessagingService` naming convention:

```
<service android:name="com.braze.push.BrazeFirebaseMessagingService" android:exported="false">
  <intent-filter>
    <action android:name="com.google.firebase.MESSAGING_EVENT" />
  </intent-filter>
</service>

```

## Issue

However when testing the push notification implementation in our Android app we found it failed. Regardless of what state the application was is (foreground/background/closed), when tapping on a push notification the app would not launch or receive data from the push. Its as if the `MESSAGING_EVENT` was not transmitted, blocked or colliding.

The best and most consistent replication of the failure was:

1. Open the application and push it to the background
2. Send and receive a new push notification
3. Tap on the notification to open the application and deep link

#### Expect outcome

App opens and receives the notification URI to deep link to.

#### Actual outcome

App crashes `D/Capacitor: App destroyed`.

## Logs

Below is a verbose log from when the push notification is recieved until `D/Capacitor: App destroyed`. As a non-android dev I'm pretty stumped at why this occurs. Any pointers, tips or advice on where to look would be great.

```
I/Braze v21.0.0 .BrazeFirebaseMessagingService: Got remote message from FCM: {a=My Message, p=0, t=My Title, _ab=true, cid=xxxxx, uri=/my-uri, ab_nc=my-channel-id}
V/Braze v21.0.0 .BrazeFirebaseMessagingService: Adding bundle item from FCM remote data with key: a and value: My Message
V/Braze v21.0.0 .BrazeFirebaseMessagingService: Adding bundle item from FCM remote data with key: p and value: 0
V/Braze v21.0.0 .BrazeFirebaseMessagingService: Adding bundle item from FCM remote data with key: t and value: My Title
V/Braze v21.0.0 .BrazeFirebaseMessagingService: Adding bundle item from FCM remote data with key: _ab and value: true
V/Braze v21.0.0 .BrazeFirebaseMessagingService: Adding bundle item from FCM remote data with key: cid and value: xxxxx
V/Braze v21.0.0 .BrazeFirebaseMessagingService: Adding bundle item from FCM remote data with key: uri and value: /my-uri
V/Braze v21.0.0 .BrazeFirebaseMessagingService: Adding bundle item from FCM remote data with key: ab_nc and value: my-channel-id
I/Braze v21.0.0 .BrazePushReceiver: Received broadcast message. Message: Intent { act=firebase_messaging_service_routing_action (has extras) }
D/Braze v21.0.0 .com.braze.Braze: Applying any pending runtime configuration values
I/Braze v21.0.0 .BrazePushReceiver: Notifications enabled
I/Braze v21.0.0 .BrazePushReceiver: Push message payload received: Bundle[{a=My Message, p=0, t=My Title, _ab=true, cid=xxxxx, uri=/my-uri, ab_nc=my-channel-id}]
D/Braze v21.0.0 .BrazeConfigurationProvider: Unable to find the xml BOOLEAN configuration value with primary key 'com_braze_in_app_message_push_test_eager_display_enabled'.Using default value 'true'.
D/Braze v21.0.0 .BrazeConfigurationProvider: Using resources value for key: 'com_braze_in_app_message_push_test_eager_display_enabled' and value: 'true'
D/Braze v21.0.0 .BrazePushReceiver: Received notification push
D/Braze v21.0.0 .BrazeNotificationUtils: Message without notification id provided in the extras bundle received. Using a hash of the message: 1508767277
V/Braze v21.0.0 .BrazePushReceiver: Creating notification with payload:

    IsPushStory = false
    IsInlineImagePush = false
    IsConversationalPush = false
    NotificationChannelId = my-channel-id
    NotificationReceivedTimestampMillis = 1674022308417
    TitleText = My Title
    ContentText = My Message
    ActionButtons = []
    PushStoryPageIndex = 0
    PushStoryPages = []
    ConversationMessages = []
    ConversationPersonMap = {}
V/Braze v21.0.0 .BrazeNotificationFactory: Using BrazeNotificationPayload:
    IsPushStory = false
    IsInlineImagePush = false
    IsConversationalPush = false
    NotificationChannelId = my-channel-id
    NotificationReceivedTimestampMillis = 1674022308417
    TitleText = My Title
    ContentText = My Message
    ActionButtons = []
    PushStoryPageIndex = 0
    PushStoryPages = []
    ConversationMessages = []
    ConversationPersonMap = {}
D/Braze v21.0.0 .BrazeNotificationUtils: Found notification channel in extras with id: my-channel-id
D/Braze v21.0.0 .BrazeNotificationUtils: Setting title for notification
D/Braze v21.0.0 .BrazeConfigurationProvider: Unable to find the xml BOOLEAN configuration value with primary key 'com_braze_push_notification_html_rendering_enabled'.Using default value 'false'.
D/Braze v21.0.0 .BrazeConfigurationProvider: Using resources value for key: 'com_braze_push_notification_html_rendering_enabled' and value: 'false'
D/Braze v21.0.0 .BrazeNotificationUtils: Setting content for notification
D/Braze v21.0.0 .BrazeNotificationUtils: Setting ticker for notification
D/Braze v21.0.0 .BrazeConfigurationProvider: Using runtime override value for key: 'com_braze_push_small_notification_icon' and value: '2131165336'
D/Braze v21.0.0 .BrazeNotificationUtils: Setting small icon for notification via resource id
D/Braze v21.0.0 .BrazeNotificationUtils: Setting large icon for notification
D/Braze v21.0.0 .BrazeNotificationUtils: Large icon bitmap url not present in extras. Attempting to use resource id instead.
D/Braze v21.0.0 .BrazeConfigurationProvider: Using runtime override value for key: 'com_braze_push_large_notification_icon' and value: '2131165335'
D/Braze v21.0.0 .BrazeNotificationUtils: Summary text not present. Not setting summary text for notification.
D/Braze v21.0.0 .BrazeNotificationUtils: Setting priority for notification
D/Braze v21.0.0 .BrazeNotificationStyleFactory: Rendering push notification with BigTextStyle
D/Braze v21.0.0 .BrazeNotificationStyleFactory: Setting style for notification
D/Braze v21.0.0 .BrazeNotificationActionUtils: No action buttons present. Not adding notification actions
D/Braze v21.0.0 .BrazeNotificationUtils: Using default accent color for notification
D/Braze v21.0.0 .BrazeConfigurationProvider: Unable to find the xml COLOR configuration value with primary key 'com_braze_default_notification_accent_color'.Using default value 'null'.
D/Braze v21.0.0 .BrazeConfigurationProvider: Using resources value for key: 'com_braze_default_notification_accent_color' and value: 'null'
D/Braze v21.0.0 .BrazeNotificationUtils: Category not present in notification extras. Not setting category for notification.
D/Braze v21.0.0 .BrazeNotificationUtils: Sending push message received broadcast
V/Braze v21.0.0 .BrazeNotificationUtils: Sending original Appboy broadcast receiver intent for RECEIVED
V/Braze v21.0.0 .BrazeNotificationUtils: Sending push action intent: Intent { act=com.my.app.intent.APPBOY_PUSH_RECEIVED }
V/Braze v21.0.0 .BrazeNotificationUtils: Sending Braze broadcast receiver intent for RECEIVED
V/Braze v21.0.0 .BrazeNotificationUtils: Sending push action intent: Intent { act=com.braze.push.intent.NOTIFICATION_RECEIVED pkg=com.my.app }
D/Braze v21.0.0 .BrazeConfigurationProvider: Unable to find the xml BOOLEAN configuration value with primary key 'com_braze_push_wake_screen_for_notification_enabled'.Using default value 'true'.
D/Braze v21.0.0 .BrazeConfigurationProvider: Using resources value for key: 'com_braze_push_wake_screen_for_notification_enabled' and value: 'true'
D/Braze v21.0.0 .BrazeNotificationUtils: Found notification channel in extras with id: my-channel-id
D/Braze v21.0.0 .BrazeNotificationUtils: Waking screen for notification
D/Braze v21.0.0 .StandaloneCoroutine: Requesting data flush on internal session close flush timer.
D/Braze v21.0.0 .a1: bo.app.m0 fired: mCommandType=ADD_REQUEST
    mBrazeEvent=null
    mSessionId=null
    mBrazeRequest=bo.app.g0@2ad5983
D/Braze v21.0.0 .a1: Triggering bo.app.m0 on 1 subscribers.
V/Braze v21.0.0 .o0: Added request to dispatcher with parameters:
    {
      "respond_with": {
        "user_id": "xxxx",
        "config": {
          "config_time": 1666846268
        }
      }
    }
V/Braze v21.0.0 .o0: SDK Auth is disabled, not adding token to request
D/Braze v21.0.0 .i0: Cache locked successfully for export: bo.app.i0@38b6974
D/Braze v21.0.0 .p6: Cache locked successfully for export: bo.app.p6@356e9d
D/Braze v21.0.0 .h3: Making request with id => "fd11fedd0a860800"
    to url: https://sdk.iad-05.braze.com/api/v3/data

    with headers:
    "Accept-Encoding" => "gzip, deflate"
    "Content-Type" => "application/json"
    "X-Braze-Api-Key" => "xxxx"

    and JSON :
    {
      "device_id": "xxxx",
      "time": 1674022309,
      "api_key": "xxxx",
      "sdk_version": "21.0.0",
      "app_version": "2.9.3",
      "app_version_code": "1100048.0.0.0",
      "attributes": [
        {
          "first_name": "xxxx",
          "email": "xxxx",
          "user_id": "xxxx"
        }
      ],
      "sdk_flavor": "cordova",
      "respond_with": {
        "user_id": "xxxx",
        "config": {
          "config_time": 1666846268
        }
      }
    }
V/Braze v21.0.0 .t: Session seal logic executing in broadcast
D/Braze v21.0.0 .t: Checking if this session needs to be sealed: c2c2ca10-c45e-406e-a004-066fbb0d48d6
I/Braze v21.0.0 .t: Session [c2c2ca10-c45e-406e-a004-066fbb0d48d6] being sealed because its end time is over the grace period. Session:
    MutableSession(sessionId=c2c2ca10-c45e-406e-a004-066fbb0d48d6, startTime=1.674022289828E9, endTime=1.674022299806E9, isSealed=false, duration=9)
D/Braze v21.0.0 .a1: bo.app.g5 fired: SessionSealedEvent(sealedSession=
    MutableSession(sessionId=c2c2ca10-c45e-406e-a004-066fbb0d48d6, startTime=1.674022289828E9, endTime=1.674022309821E9, isSealed=true, duration=19))
D/Braze v21.0.0 .a1: Triggering bo.app.g5 on 1 subscribers.
D/Braze v21.0.0 .p: Not adding session id to event: {
      "name": "se",
      "data": {
        "d": 19
      },
      "time": 1.674022309821E9,
      "session_id": "c2c2ca10-c45e-406e-a004-066fbb0d48d6"
    }
V/Braze v21.0.0 .p: Attempting to log event: {
      "name": "se",
      "data": {
        "d": 19
      },
      "time": 1.674022309821E9,
      "user_id": "xxxx",
      "session_id": "c2c2ca10-c45e-406e-a004-066fbb0d48d6"
    }
D/Braze v21.0.0 .k5: Adding event to storage with uid 72417b92-fcc2-4521-a7f6-51dd15307512
D/Braze v21.0.0 .a1: bo.app.m0 fired: mCommandType=ADD_BRAZE_EVENT
    mBrazeEvent={"name":"se","data":{"d":19},"time":1.674022309821E9,"user_id":"xxxx","session_id":"c2c2ca10-c45e-406e-a004-066fbb0d48d6"}
    mSessionId=null
    mBrazeRequest=null
D/Braze v21.0.0 .a1: Triggering bo.app.m0 on 1 subscribers.
D/Braze v21.0.0 .a1: com.braze.events.SessionStateChangedEvent fired: SessionStateChangedEvent{sessionId='c2c2ca10-c45e-406e-a004-066fbb0d48d6', eventType='SESSION_ENDED'}'
I/Braze v21.0.0 .a1: Event was published, but no subscribers were found. Saving event for later publishing to a matching subscriber. Event class: class com.braze.events.SessionStateChangedEvent
D/Braze v21.0.0 .a1: bo.app.m0 fired: mCommandType=ADD_REQUEST
    mBrazeEvent=null
    mSessionId=null
    mBrazeRequest=bo.app.g0@6620adf
D/Braze v21.0.0 .a1: Triggering bo.app.m0 on 1 subscribers.
V/Braze v21.0.0 .o0: Added request to dispatcher with parameters:
    {
      "respond_with": {
        "user_id": "xxxx",
        "config": {
          "config_time": 1666846268
        }
      }
    }
V/Braze v21.0.0 .o0: SDK Auth is disabled, not adding token to request
D/Braze v21.0.0 .i0: Received call to export dirty object, but the cache was already locked.
D/Braze v21.0.0 .p6: Received call to export dirty object, but the cache was already locked.
D/Braze v21.0.0 .o0: Event dispatched: {"name":"se","data":{"d":19},"time":1.674022309821E9,"user_id":"xxxx","session_id":"c2c2ca10-c45e-406e-a004-066fbb0d48d6"} with uid: 72417b92-fcc2-4521-a7f6-51dd15307512
D/Braze v21.0.0 .h3: Making request with id => "dee056fb97d87720"
    to url: https://sdk.iad-05.braze.com/api/v3/data

    with headers:
    "Accept-Encoding" => "gzip, deflate"
    "Content-Type" => "application/json"
    "X-Braze-Api-Key" => "xxxx"

    and JSON :
    {
      "device_id": "xxxx",
      "time": 1674022309,
      "api_key": "xxxx",
      "sdk_version": "21.0.0",
      "app_version": "2.9.3",
      "app_version_code": "1100048.0.0.0",
      "events": [
        {
          "name": "se",
          "data": {
            "d": 19
          },
          "time": 1.674022309821E9,
          "user_id": "xxxx",
          "session_id": "c2c2ca10-c45e-406e-a004-066fbb0d48d6"
        }
      ],
      "sdk_flavor": "cordova",
      "respond_with": {
        "user_id": "xxxx",
        "config": {
          "config_time": 1666846268
        }
      }
    }
V/Braze v21.0.0 .NotificationTrampolineActivity: NotificationTrampolineActivity created
V/Braze v21.0.0 .NotificationTrampolineActivity: Notification trampoline activity received intent: Intent { act=com.appboy.action.APPBOY_PUSH_CLICKED flg=0x10000000 cmp=com.my.app/com.braze.push.NotificationTrampolineActivity (has extras) }
I/Braze v21.0.0 .BrazePushReceiver: Received broadcast message. Message: Intent { act=com.appboy.action.APPBOY_PUSH_CLICKED cmp=com.my.app/com.braze.push.BrazePushReceiver (has extras) }
D/Braze v21.0.0 .com.braze.Braze: Applying any pending runtime configuration values
D/Braze v21.0.0 .BrazeNotificationUtils: Sending notification opened broadcast
V/Braze v21.0.0 .BrazeNotificationUtils: Sending original Appboy broadcast receiver intent for OPENED
V/Braze v21.0.0 .BrazeNotificationUtils: Sending push action intent: Intent { act=com.my.app.intent.APPBOY_NOTIFICATION_OPENED }
I/Braze v21.0.0 .com.braze.Braze: Logging push click. Campaign Id: xxxxx
V/Braze v21.0.0 .BrazeNotificationUtils: Sending Braze broadcast receiver intent for OPENED
V/Braze v21.0.0 .BrazeNotificationUtils: Sending push action intent: Intent { act=com.braze.push.intent.NOTIFICATION_OPENED pkg=com.my.app }
D/Braze v21.0.0 .t: Getting the stored open session
D/Braze v21.0.0 .l5: No stored open session in storage.
D/Braze v21.0.0 .p: Not adding session id to event: {
      "name": "pc",
      "data": {
        "cid": "xxxxx"
      },
      "time": 1.67402230995E9
    }
V/Braze v21.0.0 .p: Attempting to log event: {
      "name": "pc",
      "data": {
        "cid": "xxxxx"
      },
      "time": 1.67402230995E9,
      "user_id": "xxxx"
    }
D/Braze v21.0.0 .p: Publishing an internal push body clicked event for any awaiting triggers.
D/Braze v21.0.0 .a1: bo.app.c6 fired: TriggerEligiblePushClickEvent(campaignId=xxxxx, pushClickEvent={"name":"pc","data":{"cid":"xxxxx"},"time":1.67402230995E9,"user_id":"xxxx"})
D/Braze v21.0.0 .a1: Triggering bo.app.c6 on 1 subscribers.
I/Braze v21.0.0 .z0: Requesting trigger update due to trigger-eligible push click event
D/Braze v21.0.0 .a1: bo.app.m0 fired: mCommandType=ADD_REQUEST
    mBrazeEvent=null
    mSessionId=null
    mBrazeRequest=bo.app.g0@7ae2df4
D/Braze v21.0.0 .a1: Triggering bo.app.m0 on 1 subscribers.
D/Braze v21.0.0 .a1: bo.app.b6 fired: TriggerDispatchStartedEvent(request=bo.app.g0@7ae2df4)
D/Braze v21.0.0 .a1: Triggering bo.app.b6 on 1 subscribers.
V/Braze v21.0.0 .o0: Added request to dispatcher with parameters:
    {
      "respond_with": {
        "user_id": "xxxx",
        "triggers": true,
        "config": {
          "config_time": 1666846268
        }
      }
    }
V/Braze v21.0.0 .o0: SDK Auth is disabled, not adding token to request
D/Braze v21.0.0 .p: Adding push click to dispatcher pending list
D/Braze v21.0.0 .a1: bo.app.m0 fired: mCommandType=ADD_PENDING_BRAZE_EVENT
    mBrazeEvent={"name":"pc","data":{"cid":"xxxxx"},"time":1.67402230995E9,"user_id":"xxxx"}
    mSessionId=null
    mBrazeRequest=null
D/Braze v21.0.0 .a1: Triggering bo.app.m0 on 1 subscribers.
D/Braze v21.0.0 .k5: Adding event to storage with uid 7e47777f-c1d8-482a-acb1-34f863f9390a
D/Braze v21.0.0 .i0: Received call to export dirty object, but the cache was already locked.
D/Braze v21.0.0 .p6: Received call to export dirty object, but the cache was already locked.
D/Braze v21.0.0 .h3: Making request with id => "43e50a06ea14a31a"
    to url: https://sdk.iad-05.braze.com/api/v3/data

    with headers:
    "X-Braze-TriggersRequest" => "true"
    "X-Braze-DataRequest" => "true"
    "Accept-Encoding" => "gzip, deflate"
    "Content-Type" => "application/json"
    "X-Braze-Api-Key" => "xxxx"

    and JSON :
    {
      "device_id": "xxxx",
      "time": 1674022309,
      "api_key": "xxxx",
      "sdk_version": "21.0.0",
      "app_version": "2.9.3",
      "app_version_code": "1100048.0.0.0",
      "sdk_flavor": "cordova",
      "respond_with": {
        "user_id": "xxxx",
        "triggers": true,
        "config": {
          "config_time": 1666846268
        }
      }
    }
D/Braze v21.0.0 .BrazeConfigurationProvider: Using runtime override value for key: 'com_braze_handle_push_deep_links_automatically' and value: 'false'
I/Braze v21.0.0 .BrazeNotificationUtils: Not handling deep links automatically, skipping deep link handling
V/Braze v21.0.0 .NotificationTrampolineActivity: Notification trampoline activity finished processing. Delaying before finishing activity.
D/TrafficStats: tagSocket(65) with statsTag=0x539, statsUid=-1
V/Braze v21.0.0 .q5: Enabling SSL protocols: [TLSv1, TLSv1.1, TLSv1.2, TLSv1.3]
D/Braze v21.0.0 .h3: Result(id = fd11fedd0a860800 time = 239ms)
    {}
D/Braze v21.0.0 .bo.app.i6: Triggered actions Json array was null. Not de-serializing triggered actions.
D/Braze v21.0.0 .bo.app.i6: Templated message Json was null. Not de-serializing templated message.
D/Braze v21.0.0 .g0: DataSyncRequest executed successfully.
V/Braze v21.0.0 .s: Processing server response payload for user with id: xxxx
D/Braze v21.0.0 .a1: bo.app.q4 fired: RequestNetworkSuccessEvent(request=bo.app.g0@2ad5983)
D/Braze v21.0.0 .a1: Triggering bo.app.q4 on 1 subscribers.
D/Braze v21.0.0 .a1: bo.app.p0 fired: DispatchSucceededEvent(request=bo.app.g0@2ad5983)
D/Braze v21.0.0 .a1: Triggering bo.app.p0 on 1 subscribers.
V/Braze v21.0.0 .i0: Notifying confirmAndUnlock listeners for cache: bo.app.i0@38b6974
V/Braze v21.0.0 .p6: Notifying confirmAndUnlock listeners for cache: bo.app.p6@356e9d
V/Braze v21.0.0 .NotificationTrampolineActivity: Delay complete. Finishing Notification trampoline activity now
V/Braze v21.0.0 .NotificationTrampolineActivity: Notification trampoline activity paused and finishing
D/AutofillManager: Fill dialog is enabled:false, hints=[password, passwordAuto, creditCardNumber, creditCardSecurityCode, creditCardExpirationDate]
D/Capacitor: App destroyed
```

### References
- [capacitor-plugins/push-notifications/android/](https://github.com/ionic-team/capacitor-plugins/tree/main/push-notifications/android)

**Comment by radixdev** [comment]
1. Can you provide a link to the Braze dashboard campaign used to send this push?
2. As per https://www.braze.com/docs/developer_guide/platform_integration_guides/cordova/initial_sdk_setup/android/#setting-extra-configuration , can you set `android_handle_push_deep_links_automatically` to true and try to reproduce the issue?
3. What device and android API version are you testing on?
4. Can you share your AndroidManifest.xml and actual deeplink used during testing?

**Comment by Ricardinh0** [comment]
Thanks for the reply @radixdev 

1. The campaign used was a draft android push I set up. I simply targeted my user via email address, setup the notification channel id and deep link, then did a test send. Braze's Message Activity Log was registering the activity correctly.
2. I tested with `android_handle_push_deep_links_automatically` set to both `true` and `false`. The original setting pre-version-bump was `false`. Both setting made no noticeable difference to the outcome. Only the logs showed the change, but there were no obvious warnings that anything bad was happening whether true or false.
3. We've tested the notifications locally via an emulator pixel 6 android 13 and a physical device pixel 6 pro android 13.
4. The deep link was as simple as `/dashboard` which is a route within our app. This URI data is handled by capacitor on app launch via a `appUrlOpen` or `pushNotificationActionPerformed` listener. Which one depends on the applications state (foreground/background/closed).

Here's the manifest:

```
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.my.app">
  <application android:allowBackup="true" android:icon="@mipmap/ic_launcher" android:label="@string/app_name" android:roundIcon="@mipmap/ic_launcher_round" android:supportsRtl="true" android:theme="@style/AppTheme">
    <activity android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|smallestScreenSize|screenLayout|uiMode" android:name="com.my.app.MainActivity" android:label="@string/title_activity_main" android:theme="@style/AppTheme.NoActionBarLaunch" android:launchMode="singleTask" android:exported="true">
      <intent-filter>
        <action android:name="android.intent.action.MAIN" />
        <category android:name="android.intent.category.LAUNCHER" />
      </intent-filter>
      <intent-filter android:autoVerify="true">
        <action android:name="android.intent.action.VIEW" />
        <category android:name="android.intent.category.DEFAULT" />
        <category android:name="android.intent.category.BROWSABLE" />
        <data android:scheme="@string/custom_url_scheme" />
        <data android:scheme="com.my.app" />
      </intent-filter>
    </activity>
    <provider android:name="androidx.core.content.FileProvider" android:authorities="${applicationId}.fileprovider" android:exported="false" android:grantUriPermissions="true">
      <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/file_paths">
      </meta-data>
    </provider>
    <meta-data android:name="com.facebook.sdk.ApplicationId" android:value="@string/facebook_app_id" />
    <activity android:name="com.facebook.FacebookActivity" android:configChanges="keyboard|keyboardHidden|screenLayout|screenSize|orientation" android:label="@string/app_name" />
    <activity android:name="com.facebook.CustomTabActivity" android:exported="true">
      <intent-filter>
        <action android:name="android.intent.action.VIEW" />
        <category android:name="android.intent.category.DEFAULT" />
        <category android:name="android.intent.category.BROWSABLE" />
        <data android:scheme="@string/fb_login_protocol_scheme" />
      </intent-filter>
    </activity>
    <service android:name="com.braze.push.BrazeFirebaseMessagingService" android:exported="false">
      <intent-filter>
        <action android:name="com.google.firebase.MESSAGING_EVENT" />
      </intent-filter>
    </service>
  </application>
  <!-- Permissions -->
  <uses-permission android:name="android.permission.INTERNET" />
  <uses-permission android:name="android.permission.GET_ACCOUNTS" />
  <uses-permission android:name="android.permission.USE_CREDENTIALS" />
  <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
  <uses-permission android:name="com.google.android.gms.permission.AD_ID" />
  <uses-permission android:name="com.android.vending.BILLING" />
</manifest>

```

**Comment by radixdev** [comment]
I can confirm that deeplinking works as of version 2.31.0 (latest). However I have a few more questions:

1. What is the result of running `adb shell am start -a android.intent.action.VIEW -d "/dashboard" YOUR.PACKAGE.HERE` ? This tells the OS to directly attempt to open your deeplink.
2. I don't see anything in your manifest that would handle `/dashboard`. I'm assuming this might be a constant in your `strings.xml`?

Lastly, can you try adding the following to your `AndroidManifest.xml` for another test?
```
        <activity android:exported="true" android:name="com.braze.ui.activities.ContentCardsActivity" tools:replace="android:exported">
            <intent-filter android:label="@string/launcher_name">
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data android:host="deeplinking" android:scheme="juliantesting" />
            </intent-filter>
        </activity>
```

Then add `<manifest ... xmlns:tools="http://schemas.android.com/tools">` to add the `tools` namespace.
Finally, from the Braze dashboard, send a deeplink with `juliantesting://deeplinking`. The ContentCardsActivity should appear after this (verified via our public sample app to work). If this test (or Q1 from before) do not work, please send verbose logs. You can also email support@braze.com to send unfiltered logs, as that may significantly speed up the investigation.

**Comment by Ricardinh0** [comment]
Hi @radixdev

Thanks for the guidance:

1. This works. I used `adb shell am start -a android.intent.action.VIEW -d "/dashboard" com.my.app/.MainActivity`. I'm able to navigate to any route within the application from foreground, background and shutdown.
2. I guess this leads on from No.1. Route handling is dealt with by the JS app inside capacitor. In the case of the above, Capacitor lets the JS app catch the launch via an `appUrlOpen` listener and passes a `url` (`/dashboard`) via the event data. We then dispatch a route event inside the application to that URL. We don't have any routes handled within the manifest.
3. I'll give this a go today

### Final Test

This also (kinda) worked. When sending a push with `juliantesting://deeplinking` the application launched from the push interaction ( foreground, background and shutdown), but presented this:

<img width="401" alt="Screenshot 2023-01-20 at 1 51 45 pm" src="https://user-images.githubusercontent.com/3790123/213608204-8467368c-2a70-4d47-9a9c-df9c8cccd91d.png">

This content/view appears to be presented within the application, but doesn't touch the capacitor context. Also, I use chrome://inspect for capacitor logging and nothing is logged on launch from capacitor, but the application preview in the inspector switches to "The tab is inactive".

Thanks again for probing this issue.

**Comment by Ricardinh0** [comment]
Hey @radixdev, closing this as I boiled it down to the capacitor v4 upgrade. I tested the `appboy-cordova-sdk` up to `2.31.0` with Cap v3 and it worked as expected within our setup.

The issue was that Capacitor v4 now has no requirement to continue using `jcenter()` in its `build.gradle`. This is because they're now using the latest Cordova Android version, hosted on Maven Central. Therefore `jcenter()` is suggested as an optional removal within the upgrade. However, the optional nature of that depends on whether other plugins or native dependencies still need it, and it appears the `appboy-cordova-sdk` still needs it.

Thank you for helping me diagnose this issue 🙌
